src/logic/calculate.js

```diff
@@ -0,0 +1,36 @@
+import operate from './operate';
+
+const calculate = (data, name) => {
```

Comment:
### [Optional]
For good naming conversion, it would be nice to make use of `buttonName` instead of `name` so that the naming can be related to the logic, which is to get the name of the button being clicked right.